src/context/video-listing-context.js

```diff
@@ -0,0 +1,43 @@
+import { createContext, useContext, useEffect, useReducer } from 'react';
+import { getVidoes } from '../services';
+
+import { videoListReducer } from '../reducers';
+
+const VideoListingContext = createContext();
+
+const VideoProvider = ({ children }) => {
+    const [videoListState, videoListDispatch] = useReducer(videoListReducer, {
+        data: [],
+    });
+
+    useEffect(
+        () =>
+            (async () => {
+                try {
+                    const {
+                        data: { videos },
+                        status,
+                    } = await getVidoes();
+                    if (status === 200) videoListDispatch({ type: 'LOAD_VIDEOS', payload: videos });
+                    else throw new Error('Falied with code: ', status);
+                } catch {
+                    throw new Error('Error fetching data.');
```

Comment:
catch will catch error anyway, so you don't need to throw an error explicitly

Reply:
Done